api/agent/agent.go

```diff
@@ -1321,6 +1299,81 @@ func (c *container) WriteStat(ctx context.Context, stat drivers.Stat) {
 	c.swapMu.Unlock()
 }
 
+// SetEvictable sets the container evictability.
+func (c *container) SetEvictable(call *call, evictable bool) {
+	c.evictLock.Lock()
+	defer c.evictLock.Unlock()
+
+	if evictable {
+		if c.evictToken == nil {
+			c.evictToken = c.evictor.CreateEvictToken(call.slotHashId, call.Memory+uint64(call.TmpFsSize), uint64(call.CPUs))
+		}
+		c.evictToken.SetEvictable(true)
+		return
+	}
+
+	if c.evictToken == nil {
+		return
+	}
+
+	c.evictToken.SetEvictable(false)
+
+	// are we too late?
+	select {
+	case <-c.evictToken.C:
+	default:
+		return
+	}
+
+	// delete this already evicted token and fetch a new one
+	c.evictor.DeleteEvictToken(c.evictToken)
+	c.evictToken = c.evictor.CreateEvictToken(call.slotHashId, call.Memory+uint64(call.TmpFsSize), uint64(call.CPUs))
```

Comment:
Was about to remark that it would be worthwhile to `SetEvictable(false)` after creation, but then I realized that this is go.

Reply:
By default eviction tokens are not in evictable state. This allows for more flexible usage avoiding some race conditions.